test/resources/knativeserving.go

```diff
@@ -143,3 +123,38 @@ func getTestKSOperatorCRSpec() v1alpha1.KnativeServingSpec {
 		},
 	}
 }
+
+func configureIngressClass(ks *v1alpha1.KnativeServing) error {
+	if ks == nil {
+		return fmt.Errorf("ks is nil, cannot configure IngressClass")
+	}
+
+	ingressClass := "istio.ingress.networking.knative.dev"
+	if ingressClassOverride := os.Getenv("INGRESS_CLASS"); ingressClassOverride != "" {
+		ingressClass = ingressClassOverride
+	}
+	var istioEnabled, contourEnabled, kourierEnabled bool
+	switch strings.Split(ingressClass, ".")[0] {
+	case "istio":
+		istioEnabled = true
+	case "contour":
+		contourEnabled = true
+	case "kourier":
+		kourierEnabled = true
+	}
+
+	if !istioEnabled {
+		(*ks).Spec.Ingress = &v1alpha1.IngressConfigs{
```

Comment:
```suggestion
		ks.Spec.Ingress = &v1alpha1.IngressConfigs{
```

Not sure what the indirection is about here :thinking: (this applies to all occurences below)

Reply:
Thanks for the comment. Yeah, I am still trying to get myself familiar with Go. I will change the code based on the suggestion.